Mastermind/MastermindService.cs

```diff
@@ -0,0 +1,51 @@
+using Mastermind.Domain.Models;
+using Mastermind.Presentation.InputOutput;
+using Mastermind.Randomizer;
+
+namespace Mastermind;
+
+public class MastermindService
+{
+    private readonly Controller _controller;
+    private readonly Game _game;
+    
+    public MastermindService(IInputOutput inputOutput)
+    {
+        _game = new Game();
+        _controller = new Controller(inputOutput);
+    }
+
+    public void PlayGame(IRandomizer randomizer)
+    {
+        _controller.DisplayInitialMessage();
+        
+        var gameChecker = new Domain.BusinessRules.GamePlay(randomizer, _game);
+        gameChecker.Initialise();
+        var gameStatus = GameStatus.Playing;
+
+        while (gameStatus == GameStatus.Playing)
+        {
+            _controller.UpdatePlayerGuess();
+
+            if (_controller.HasPLayerQuit())
+            {
+                gameStatus = GameStatus.Quit;
+            }
+            else
+            {
+                var userGuess = _controller.GetUpdatedUserGuess();
+                gameChecker.EvaluatePredictedAnswer(userGuess);
```

Comment:
The game loop can still be tightened up a bit. Ideally, we'd be able to remove all of these high-level if statements and differ that responsibility to the low-level business rules. Additionally, the service should only be making calls to the controller and the view (because it's acting as a boundary between the two layers)